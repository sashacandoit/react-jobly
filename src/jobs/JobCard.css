.JobCard {
  min-height: 22vh;
}