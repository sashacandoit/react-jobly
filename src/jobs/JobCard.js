import React from "react";
import { Link } from "react-router-dom";
import Card from '@mui/material/Card';
import CardMedia from '@mui/material/CardMedia';
import CardContent from '@mui/material/CardContent';
import Typography from '@mui/material/Typography';

const JobCard = ({ id, title, salary, equity, company, companyHandle }) => {
  return (
    <Card sx={{ minWidth: 25 }}>
      <CardContent>
        {/* <CardMedia
          component="img"
          height="150"
          image={company.logoUrl}
          alt={company.name}
        /> */}
        <Typography variant="h5" component="div">
          {title}
        </Typography>

        <Typography variant="body2">
          {salary}
        </Typography>
        
        <Link to={`/companies/${companyHandle}`}>
          <Typography variant="body2">
            {company}
          </Typography>
        </Link>
        
      </CardContent>
    </Card>
  )
}

export default JobCard;