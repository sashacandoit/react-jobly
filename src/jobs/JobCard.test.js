import React from "react";
import { render } from "@testing-library/react";
import JobCard from "./JobCard";
import { MemoryRouter } from "react-router";

it("matches snapshot", function () {
  const { asFragment } = render(
    <MemoryRouter>
      <JobCard
        id="1"
        title="Test Job"
        salary="1000"
        companyName="Test Company"
      />
    </MemoryRouter>,
  );
  expect(asFragment()).toMatchSnapshot();
});

