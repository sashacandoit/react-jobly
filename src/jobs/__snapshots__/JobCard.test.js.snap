// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`matches snapshot 1`] = `
<DocumentFragment>
  <div
    class="MuiPaper-root MuiPaper-elevation MuiPaper-rounded MuiPaper-elevation1 MuiCard-root css-x1r8w7-MuiPaper-root-MuiCard-root"
  >
    <div
      class="MuiCardContent-root css-46bh2p-MuiCardContent-root"
    >
      <div
        class="MuiTypography-root MuiTypography-h5 css-ag7rrr-MuiTypography-root"
      >
        Test Job
      </div>
      <p
        class="MuiTypography-root MuiTypography-body2 css-e784if-MuiTypography-root"
      >
        1000
      </p>
      <a
        href="/companies/undefined"
      >
        <p
          class="MuiTypography-root MuiTypography-body2 css-e784if-MuiTypography-root"
        />
      </a>
    </div>
  </div>
</DocumentFragment>
`;
